muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/records.tsx

```diff
@@ -104,188 +90,170 @@ function getActivity(props: RecordsProps, workspace: WorkspaceType){
-                        <span className="application-list__header-icon icon-books"></span>
-                        <span className="application-list__header-primary">{workspace.name} {workspace.nameExtension ? "(" + workspace.nameExtension + ")" : null}</span> 
+                {record.groupCurriculumIdentifier ? <div onClick={this.sortWorkspaces.bind(this, record.workspaces)} className="application-list__header-container application-list__header-container--sorter">
+                  <h3 className="application-list__header application-list__header--sorter">{storedCurriculumIndex[record.groupCurriculumIdentifier]}</h3>
+                  <div className={`icon-sort-alpha-${this.state.sortDirectionWorkspaces === 'asc' ? 'desc' : 'asc'}`}></div>
+                </div> : null}
+                {record.workspaces.map((workspace)=>{
+                  // Do we want an special way to display all these different states? passed is very straightforward but failed and
+                  // incomplete might be difficult to understand
+                  let extraClassNameState = "";
+                  if (workspace.studentAssessmentState.state === "pass"){
+                    extraClassNameState = "state-PASSED"
+                  } else if (workspace.studentAssessmentState.state === "fail"){
+                    extraClassNameState = "state-FAILED"
+                  } else if (workspace.studentAssessmentState.state === "incomplete"){
+                    extraClassNameState = "state-INCOMPLETE"
+                  }
+                  return <ApplicationListItem className={`course course--studies ${extraClassNameState}`} key={workspace.id} onClick={this.goToWorkspace.bind(this, user, workspace)}>
+                    <ApplicationListItemHeader modifiers="course" key={workspace.id}>
+                      <span className="application-list__header-icon icon-books"></span>
+                      <span className="application-list__header-primary">{workspace.name} {workspace.nameExtension ? "(" + workspace.nameExtension + ")" : null}</span>
+                      <div className="application-list__header-secondary">
                         {getEvaluationRequestIfAvailable(this.props, workspace)}
                         {getAssessments(this.props, workspace)}
                         {getActivity(this.props, workspace)}
+                      </div>
+                    </ApplicationListItemHeader>
+                  </ApplicationListItem>
+                })}
+                {record.transferCredits.length ?
+                  <div className="application-list__header-container application-list__header-container--sorter" onClick={this.sortRecords.bind(this, record.transferCredits)}>
+                    <h3 className="application-list__header application-list__header--sorter">{this.props.i18n.text.get("plugin.records.transferCredits")} ({storedCurriculumIndex[record.groupCurriculumIdentifier]})</h3>
```

Comment:
Transfer credits may or may not have a curriculum identifier. When curriculum identifier is not available, this code renders empty brackets () after the transfer credits title.